src/storage/mvcc/txn.rs

```diff
@@ -533,6 +533,41 @@ impl<S: Snapshot> MvccTxn<S> {
         Ok(is_pessimistic_txn)
     }
 
+    pub fn refresh_lock(&mut self, key: Key, options: &Options) -> Result<u64> {
+        if let Some(lock) = self.reader.load_lock(&key)? {
+            if lock.ts != self.start_ts {
+                // locked by another transaction
+                info!(
+                    "locked by another transaction!";
+                    "previous" => self.start_ts,
+                    "current" => lock.ts,
+                );
+
+                return Err(Error::KeyIsLocked {
+                    key: key.to_raw()?,
+                    primary: lock.primary,
+                    ts: lock.ts,
+                    ttl: lock.ttl,
+                    txn_size: options.txn_size,
+                });
+            }
+            // does not consider pessimistic lock for now.
+            self.lock_key(key, lock.lock_type, lock.primary, lock.short_value, options);
+            Ok(options.lock_ttl)
```

Comment:
emmm, I think it is useless to return lock's ttl.

Reply:
It should be returned through the RPC Response

Reply:
I found he didn't use it and we don't have a `StorageCb` that can send back a TTL.😂

Reply:
Oh, I'm wrong. It's not used.